### PR TITLE
plumb kvstore to the adapter

### PR DESCRIPTION
**Pull request by vaikas**

### CODE REVIEW

File: pkg/reconciler/vsphere/resources/configmap.go~ [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
```
Comment (mattmoor) [inline pkg/reconciler/vsphere/resources/configmap.go~]: LOL you need your `.gitignore` 🤣

File: pkg/reconciler/vsphere/resources/names/names.go [hunk 2]
```diff
@@ -29,3 +29,9 @@ func DeploymentName(vms *v1alpha1.VSphereSource) string {
 func SinkBindingName(vms *v1alpha1.VSphereSource) string {
 	return kmeta.ChildName(vms.Name, "sinkbinding")
 }
+func ConfigMapName(vms *v1alpha1.VSphereSource) string {
+	return kmeta.ChildName(vms.Name, "configmap")
+}
+func RoleBindingName(vms *v1alpha1.VSphereSource) string {
+	return kmeta.ChildName(vms.Name, "rolebinding")
+}
```
Comment (mattmoor) [inline pkg/reconciler/vsphere/resources/names/names.go]:
```suggestion

func ConfigMapName(vms *v1alpha1.VSphereSource) string {
	return kmeta.ChildName(vms.Name, "configmap")
}

func RoleBindingName(vms *v1alpha1.VSphereSource) string {
	return kmeta.ChildName(vms.Name, "rolebinding")
}
```

File: pkg/reconciler/vsphere/resources/role_binding.go~ [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (mattmoor) [inline pkg/reconciler/vsphere/resources/role_binding.go~]: same

File: pkg/vsphere/adapter.go [hunk 4]
```diff
@@ -69,13 +74,28 @@ func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClie
...
 
+	kubeclient := ctx.Value(client.Key{})
+	if kubeclient == nil {
+		logger.Fatalf("NIL kubeclient")
+	}
+
+	// TODO: configmap name needs to be passed in the env, or the name of the source needs to
+	// plumbed in so we can derive it here.
+	// https://github.com/mattmoor/vmware-sources/issues/8
+	kvstore := kvstore.NewConfigMapKVStore(ctx, "test", env.Namespace, kubeclient.(*kubernetes.Clientset).CoreV1())
```
Comment (mattmoor) [inline pkg/vsphere/adapter.go:85]: What's with the type cast?

Comment (vaikas) [reply pkg/vsphere/adapter.go:85]: Value returns interface{} if I don't cast it, I get this:
./adapter.go:85:79: kubeclient.CoreV1 undefined (type interface {} is interface with no methods)

🤷‍♂

Comment (mattmoor) [reply pkg/vsphere/adapter.go:85]: Oh, you are using `context.Value` not the injection stuff. 🤦‍♂ 

I'll add it to my shit list.

Comment (mattmoor) [reply pkg/vsphere/adapter.go:85]: actually you are using the key, so `kubeclient.Get(ctx)` should work here and it does the cast for you.

Comment (mattmoor) [reply pkg/vsphere/adapter.go:85]: LOL but you are using the key for this repo's client, not the kubernetes one 🤣

**Pull request merged**